server/src/main/java/org/apache/druid/server/http/CompactionConfigUpdateRequest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.http;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.indexer.CompactionEngine;
+
+import javax.annotation.Nullable;
+
+/**
+ * Payload to update the cluster-level compaction config.
+ * All fields of this class must be nullable. A non-value indicates that the
+ * corresponding field is being updated.
+ */
+public class CompactionConfigUpdateRequest
+{
+  private final Double compactionTaskSlotRatio;
+  private final Integer maxCompactionTaskSlots;
+  private final Boolean useAutoScaleSlots;
+  private final CompactionEngine compactionEngine;
+
+  @JsonCreator
+  public CompactionConfigUpdateRequest(
```

Comment:
It would be. I think we should also rename `CoordinatorCompactionConfig` to `ClusterCompactionConfig` or `GlobalCompactionConfig`. What do you think?

Also, would it be okay to do the rename changes and the API deprecation in a follow up PR?
(I have some more follow-up changes lined up).

Reply:
Yes, particularly since it would no longer reside with the coordinator. Among the 2, I would prefer `ClusterCompactionConfig`

Reply:
Cool, I guess we should rename the API to `/cluster` as well then.